### DISCUSSION TITLE
Method to delay Cluster deployment until after the MutatingWebhookConfiguration resource is up?

### DISCUSSION THREAD

**Discussion by vanakema** [cloudnative-pg/cloudnative-pg#2117] [Q&A]
Hey there, 

I've been trying to make a POC using cloudnative-pg in cdk8s's typescript generator, and we have the yaml files successfully compiling and there doesn't seem to be anything wrong with the way the files are being created. The issue I have found is that the webhook endpoint that is hit as part of standing up the Cluster resource fails. I know that there isn't any issue with the code because if you apply the manifest a couple seconds later, it succeeds to deploy, I assume due to the endpoint actually being accessible now.

This is the full error:
```
Error from server (InternalError): error when creating "foo-common.k8s.yaml": Internal error occurred: failed calling webhook "mcluster.kb.io": failed to call webhook: Post "https://cnpg-webhook-service.cnpg-system.svc:443/mutate-postgresql-cnpg-io-v1-cluster?timeout=10s": dial tcp 10.106.161.31:443: connect: connection refused
```

The endpoint is `https://cnpg-webhook-service.cnpg-system.svc:443/mutate-postgresql-cnpg-io-v1-cluster?timeout=10s`

I've tried to figure out how we could make the Cluster either keep retrying its deployment until it succeeds or fail after a certain amount of time, but all of the k8s native methods don't seem possible (initContainer that retries the endpoint until it succeeds, and healthProbe) since those are things put on Deployment resources.

I admit I am very much green to k8s (I'm at that point where I am relatively comfortable with `kubectl`, but am also acutely aware of just how little I know), so I'm hoping someone has a clever way around this issue, or maybe an obvious way that I just haven't discovered yet.

I am almost 100% certain that there is not a weird issue with cdk8s and the way it is bringing the operator manifest in, since I did a diff of the outputted yaml manifest, and the only real change was the addition of my Cluster resource, which is below

```yaml
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name: cluster-example
spec:
  instances: 3
  primaryUpdateStrategy: unsupervised
  storage:
    size: 1Gi
```

I'd really appreciate any guidance on this problem, and thank you in advance for your response!

**Comment by sxd** [comment]
Hi @vanakema 

I'm converting this into a discussion since it's more a questions related to k8s rather than an issue with the operator.

Regards!

**Reply by saitejaboggarapu** [reply]
@sxd is there any solution for this?

**Comment by wolframm** [comment]
`kubectl rollout status deployment cnpg-controller-manager -n cnpg-system --timeout=90s` seems to do the trick.
